### PR TITLE
Add `css_paths` and `head_paths` parameters

### PR DESCRIPTION
**Pull request by abidlabs**
Based on @hysts's feedback [internally](https://huggingface.slack.com/archives/C07N8RBSR39/p1727687620544439), modifying the approach we are taking to providing paths for static resources

Open to feedback

### CODE REVIEW

File: gradio/utils.py [hunk 1]
```diff
@@ -1544,3 +1544,11 @@ def get_node_path():
         "If you wish to use the node backend, please install node 18 and/ or set the path with the GRADIO_NODE_PATH environment variable."
     )
     return None
+
+
+def none_or_singleton_to_list(value: Any) -> list:
```
Comment (abidlabs) [inline gradio/utils.py:1549]: We use this pattern quite a bit in the library so added a helper function

**Comment by dawoodkhan82** [comment]
It doesn't seem like adding multiple file paths in `js_paths` seems to work. ex: `js_paths=["test.js", "test2.js]` where each file contains a simply js function. This might be true for `css_paths` and `html_paths` too, but didn't test.

Specifying one js file (not in a list) works fine btw.

**Comment by abidlabs** [comment]
Updated the PR to remove `js_paths` -- `css_paths` and `head_paths` should work as expected. Here's a quick test for `head_paths`:

```py
import gradio as gr

head1 = "<script>alert('hi')</script>"

with open("head1.html", "w") as file:
    file.write(head1)

with gr.Blocks(js="alert('hello')", head_paths=["head1.html"]) as demo:
    pass

demo.launch(ssr_mode=False)  # head, css currently don't work on ssr mode
```

Thanks @dawoodkhan82 for the review and @pngwn for the feedback

**Comment by dawoodkhan82** [comment]
Nice, tested everything and it works great. However it doesn't seem to work on frontend dev mode for me. Had to use the built version of the gradio frontend. This might just be because I'm not properly disabling the ssr version?

**Review by dawoodkhan82 — APPROVED** [review]

**Comment by abidlabs** [comment]
Thanks for reviewing @dawoodkhan82!

> Nice, tested everything and it works great. However it doesn't seem to work on frontend dev mode for me. Had to use the built version of the gradio frontend. This might just be because I'm not properly disabling the ssr version?

That's strange, since this PR doesn't make any frontend changes. I agree that its probably due to ssr, which is being addressed in a separate PR.

**Pull request merged**